src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -2803,14 +2821,21 @@ private void EmitFieldStore(BoundFieldAccess fieldAccess)
                 _builder.EmitOpCode(ILOpCode.Volatile);
             }
 
-            _builder.EmitOpCode(field.IsStatic ? ILOpCode.Stsfld : ILOpCode.Stfld);
-            EmitSymbolToken(field, fieldAccess.Syntax);
+            if (field.RefKind != RefKind.None && !refAssign)
+            {
+                //NOTE: we should have the actual field already loaded, 
+                //now need to do a store to where it points to
+                EmitIndirectStore(field.Type, fieldAccess.Syntax);
```

Comment:
>EmitIndirectStore(field.Type, fieldAccess.Syntax); [](http://example.com/codeflow?start=16&length=50)

It looks like we are applying volatile semantics to the field as well as to the underlying value. Is this intentional? Perhaps specification should mention that explicitly. Based on the placement of the required custom modifier, it applies to the underlying value rather than to the ref.

Reply:
The spec has been updated to include the following, and the test plan includes an item to disallow `volatile ref` fields.
> A `ref` field cannot be declared `static`, `volatile` or `const`